### PR TITLE
update session caching

### PR DESCRIPTION
**Pull request by jomartin-999**
The session caching included a lot of duplicate information and was not clearly organized.

This update refactors the page to be more concise and clear about the instructions and steps for using session caching.

### CODE REVIEW

File: content/session-caching.html.md.erb [hunk 1]
```diff
@@ -18,113 +18,130 @@ title: Tomcat Session State Caching
...
+When deploying applications with session state caching, ensuring compatibility between the versions of Tomcat, GemFire, and the geode-store is crucial for optimal performance and reliability.
 
-When the `session-replication` tag is specified,
-the sessions will be stored in the <%=vars.product_name_short%> service instance,
-thus moving the data out of Tomcat.
-An application bound to this service will start running in a Tomcat instance.
-A region for storing sessions will be created.
-The region will be named `gemfire_modules_sessions`.
-A 30-minute expiration for sessions will be automatically configured.
+>Confirm that you are using a Java buildpack version 4.30 or later.
```
Comment (pnikonowicz) [inline content/session-caching.html.md.erb:30]: i would consider just removing this. we discovered that there is a significant chance that the customer will need to use a custom build pack. 

if we decide to keep this line, the fist question i would have is "how do we do this"?

File: content/session-caching.html.md.erb [hunk 2]
```diff
@@ -18,113 +18,130 @@ title: Tomcat Session State Caching
...
     $ cf update-service new-service-instance -t session-replication
     </pre>
 
-- Option 3: When creating the service,
-include the substring `session-replication` anywhere within the
-service instance name.
-An example with the substring is `my-service-instance-session-replication`.
-
-## <a id="java-ssc-buildpack"></a> Java Buildpack Version Compatibility with Tomcat Session State Caching
+- Option 3: Include session-replication anywhere in the service instance name when creating the service:
```
Comment (pnikonowicz) [inline content/session-caching.html.md.erb:52]: can we deprecate this?

File: content/session-caching.html.md.erb [hunk 3]
```diff
@@ -484,12 +438,9 @@ to the `applications` section of the `manifest.yml` file:
     cf push -f ./manifest.yml --no-start -b BUILDPACK-NAME
     ```
 
-    where `BUILDPACK-NAME` is the name you chose for your custom buildpack,
-    or it is the URL of the most recent Java buildpack version,
-    if you did not create a custom buildpack.
+   Replace `BUILDPACK-NAME` with the name of your custom buildpack or the URL of the most recent Java buildpack version if you did not create a custom buildpack.
```
Comment (rishipathak10) [inline content/session-caching.html.md.erb:441]: Does recommending the customer to use  the most recent java buildpack make sense . The application might fail if custom buildpack with a compatiple geode store is not used.

File: content/session-caching.html.md.erb [hunk 4]
```diff
@@ -255,67 +238,57 @@ with the string to change in bold:
 
     <pre>
     geode_store:
-      <strong>version: 1.13.+</strong>
+      <strong>version: 1.14.9</strong>
```
Comment (rishipathak10) [inline content/session-caching.html.md.erb:241]: java buildpack 4.70 already uses geode store version 1.14.9. If we are guiding the customer to create a customer buildpack, I think we should use the more recent versions of geode-store  like 2.0.2/20.3/2.1.0

File: content/session-caching.html.md.erb [hunk 5]
```diff
@@ -136,90 +153,57 @@ Select and implement one of these three ways.
...
-          your application's `manifest.yml` file.
-          For details about specifying configuration values with an environment variable in the
-         Java Buildpack, see [Configuration and Extension](https://github.com/cloudfoundry/java-buildpack/blob/main/README.md#configuration-and-extension) in the Cloud Foundry Java Buildpack repository in GitHub.
-        You can find the buildpack's <code>geode_store</code> version in the <code>/config/tomcat.yml</code> file. For example:
+        1. Set the `geode_store` version environment variable in your application's `manifest.yml` file.  For example:
 
             ```
             env:
-              JBP_CONFIG_TOMCAT: '{ geode_store: { version: 1.13.0 } }'
+              JBP_CONFIG_TOMCAT: '{ geode_store: { version: 1.14.9 } }'
```
Comment (rishipathak10) [inline content/session-caching.html.md.erb:178]: java buildpack 4.70 already uses geode store version 1.14.9. If we are guiding the customer to create a custom buildpack, I think we should use the more recent versions of geode-store like 2.0.2/20.3/2.1.0

File: content/session-caching.html.md.erb [hunk 6]
```diff
@@ -18,113 +18,130 @@ title: Tomcat Session State Caching
...
+- [Create a Custom Buildpack](#custom-ssc-buildpack) and modify the version of Tomcat in your custom buildpack.
+
+- Specify the online buildpack in the app's `manifest.yml` file, and set an environment variable in the app's `manifest.yml` file as described in the
+[Java Buildpack Configuration and Extension README](https://github.com/cloudfoundry/java-buildpack/blob/main/README.md#configuration-and-extension).
+
+For example:
+
+    ```
+    buildpacks:
+      - https://github.com/cloudfoundry/java-buildpack.git#v4.37
```
Comment (rishipathak10) [inline content/session-caching.html.md.erb:82]: Would it make sense to mention the latest version of java buildpack here even though this  is just an example?

File: content/session-caching.html.md.erb (hunk 1 above)
Comment (jomartin-999) [reply content/session-caching.html.md.erb:30]: Are you saing to remove the note about using a buildpack > 4.30? And you're follow up question is how to help customers determine what version of the buildpack they're using?

File: content/session-caching.html.md.erb (hunk 2 above)
Comment (jomartin-999) [reply content/session-caching.html.md.erb:52]: Yes please. We have to support it until the next major, so we should also file a ticket to remove it in the next major, so we don't forget.

File: content/session-caching.html.md.erb (hunk 3 above)
Comment (jomartin-999) [reply content/session-caching.html.md.erb:441]: Thinking longer term, I'm hoping we can get back to a place where the default buildpack will work. At some point we'll need to update the geode-store to 2.x.

**Comment by rishipathak10** [comment]
Hello @jomartin-999 @pnikonowicz I think when we recommend to use the custom java-buildpack in the doc and say that geode-store version should match the Gemfire Tile version, customer might still be confused about the java-buildpack version to use. Is it reasonable to say that customer uses the same java buildpack version he has on TAS and configure it  to use the geode-store version which mathces the Gemfire Tile version?

**Review by pnikonowicz — APPROVED** [review]

**Pull request merged**